cleanlab/datalab/issue_manager/outlier.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, ClassVar, Dict, Optional, Union, cast
+
+from scipy.sparse import csr_matrix
+from scipy.stats import iqr
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+
+from cleanlab.datalab.issue_manager import IssueManager
+from cleanlab.datalab.knn import KNN
+from cleanlab.outlier import OutOfDistribution, transform_distances_to_scores
+
+if TYPE_CHECKING:  # pragma: no cover
+    from cleanlab.datalab.datalab import Datalab
+
+
+class OutOfDistributionIssueManager(IssueManager):
+    """Manages issues related to out-of-distribution examples."""
+
+    description: ClassVar[
+        str
+    ] = """An outlier issue refers to examples that are very different
+        from the rest of the dataset (i.e. potentially out-of-distribution).
+
+        Training/evaluating ML models with such examples may have unexpected consequences.
+
+        Examples may be considered as outliers if they:
+            - Are drawn from different distributions than the rest of the dataset.
+            - Are rare or anomalous events with extreme values.
+            - Have measurement- or data-collection errors.
+            - etc.
```

Comment:
```suggestion
    ] = """Examples that are very different from the rest of the dataset
          (i.e. potentially out-of-distribution or rare/anomalous instances).
```

Reply:
recommend saving the rest of this description for the docs page / blog. Not in the report itself.